sqlmesh/core/model/kind.py

```diff
@@ -208,8 +208,36 @@ def to_expression(self, dialect: str = "", **kwargs: t.Any) -> d.ModelKind:
 class IncrementalByUniqueKeyKind(_Incremental):
     name: Literal[ModelKindName.INCREMENTAL_BY_UNIQUE_KEY] = ModelKindName.INCREMENTAL_BY_UNIQUE_KEY
     unique_key: t.List[exp.Expression]
+    when_matched: t.Optional[exp.When] = None
+
     _unique_key_validator = _unique_key_validator
 
+    @field_validator("when_matched", mode="before")
+    @classmethod
+    def _when_matched_validator(cls, v: t.Optional[exp.When]) -> t.Optional[exp.When]:
+        def replace_table_references(expression: exp.Expression) -> exp.Expression:
+            from sqlmesh.core.engine_adapter.base import (
```

Comment:
do we need to set the dialect meta here? look at other places where we validate expressions

Reply:
Do you know why the parser doesn't automatically do this when returning expressions?

Reply:
Also not sure what the purpose of this is and the impact if it is not included.

Reply:
I did add it.

Reply:
parser doesn't add it to every node automatically